core/src/env/engine/on_chain/ext.rs

```diff
@@ -79,27 +154,37 @@ mod sys {
         );
         pub fn ext_terminate(beneficiary_ptr: u32, beneficiary_len: u32) -> !;
 
-        pub fn ext_dispatch_call(call_ptr: u32, call_len: u32);
-
-        pub fn ext_scratch_size() -> u32;
-        pub fn ext_scratch_read(dst_ptr: u32, offset: u32, len: u32);
-        pub fn ext_scratch_write(src_ptr: u32, len: u32);
+        pub fn ext_call_chain_extension(
+            func_id: u32,
+            input_ptr: u32,
+            input_len: u32,
+            output_ptr: u32,
+            output_len_ptr: u32,
+        ) -> u32;
```

Comment:
Having this in the set of `ext_` calls won't invalidate compiled contracts as long as they do not use this functionality. We could simply `cfg` it until it has been properly implemented on the Substrate side. Or we could just leave it as it is so that ink! magically already supports the whole thing as soon as it is implemented on Substrate.